src/foam/core/AbstractBooleanPropertyInfo.java

```diff
@@ -12,4 +12,8 @@
   public int compareValues(boolean b1, boolean b2) {
     return Boolean.compare(b1, b2);
   }
+  
+  public int comparePropertyValue(Object key, FObject o) {
+    return compareValues((boolean) key, (boolean) f(o));
```

Comment:
This is very expensive because f() is going to create an object for each comparison. Which if inserting or finding and millions objects a second would be 20M objects/s. Instead of f(), calling get_() wouldn't create an object.